### PR TITLE
[KRYS-3855] Change api chainID logic

### PR DESCRIPTION
**Pull request by MrQKyber**

### CODE REVIEW

File: KyberNetwork/KyberNetwork/Networking/External/ChainType.swift [hunk 1]
```diff
@@ -112,6 +112,16 @@ enum ChainType: Codable, CaseIterable {
     }
   }
   
+  func getChainIdParam() -> [String] {
+    if self == .all {
+      return ChainType.getAllChainID().map { e in
+        return "\(e)"
+      }
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]:
```
return ChainType.getAllChainID().map { chainID in
        return "\(chainID)"
      }
```
Please don't use variable name like 'a, b, c, e, x, i`

File: KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift [hunk 2]
```diff
@@ -146,13 +146,21 @@ class KNGeneralProvider {
   var chainAddressType: KAddressType {
     return currentChain.addressType
   }
+  
+  var overviewCurrentChain: ChainType
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift:150]: I think `overviewCurrentChain` should be a property of Overview module, not here

File: KyberNetwork/KyberNetwork/Networking/External/ChainType.swift (hunk 1 above)
Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: e, x ,I is quite popular 🤔.

File: KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift (hunk 2 above)
Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift:150]: but we need access it in other place like balance coordinator, ... etc

File: KyberNetwork/KyberNetwork/Networking/External/ChainType.swift (hunk 1 above)
Comment (tungnguyen20) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: Please provide some examples, I don't think a good project contains variable names like that

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: for I in array, for e in array ... it's so popular

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: we have some shorter format like "\($0)" . It depend on current context to use

Comment (tungnguyen20) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: for i, for e is only for education. We are working on a large project. 
If you use short hand $0, I'm okay. But if you use a name, please choose a meaning one.

File: KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift (hunk 2 above)
Comment (tungnguyen20) [reply KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift:150]: If BalanceCoordinator only use to load balance on the overview screen, please make it as a property of Overview, not a AppCoordinator property

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/KNGeneralProvider.swift:150]: Can not understand your idea, plz help me to fix it. I am busy with other task

File: KyberNetwork/KyberNetwork/Networking/External/ChainType.swift (hunk 1 above)
Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: may be we will discuss coding style in other meeting.

Comment (toidv) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: What is the problem to have a better name @MrQKyber

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Networking/External/ChainType.swift:119]: yes, it's quick to change but it is not an issue or error

**Review by tungnguyen20 — APPROVED** [review]

**Pull request merged**